app/domain/authentication/resource_restrictions/fetch_non_empty_resource_annotations.rb

```diff
@@ -0,0 +1,46 @@
+require 'command_class'
+
+module Authentication
+  module ResourceRestrictions
+
+    FetchNonEmptyResourceAnnotations = CommandClass.new(
```

Comment:
The name is confusing. Class does not fetch non empty annotations purely rather throws exception on empty annotation...

Reply:
You are right. But i cant think of better names. Any suggestions?

Reply:
i think it can still be `FetchResourceAnnotations` and raise an error in case an annotation is empty. For example, the `Authenticate` class does not allow wrong credentials and it is not called `AuthenticateCorrectCredentials`. The correct credentials is an implementation detail, the same as the non-empty annotations.